cluster/src/dunit/scala/io/snappydata/cluster/SnappySecureJob.scala

```diff
@@ -0,0 +1,243 @@
+      if (jobConfig.getString(opCode).equalsIgnoreCase("sqlOps")) {
+        createPartitionedRowTableUsingSQL(snSession)
+        createPartitionedRowTableUsingAPI(snSession)
+      } else {
+        accessAndModifyTablesOwnedByOthers(snSession, jobConfig)
+      }
+      pw.println(msg)
+    } finally {
+      pw.close()
+    }
+    msg
+  }
+
+  override def isValidJob(sc: SnappySession, config: Config): SnappyJobValidation = {
+    SnappyStreamingSecureJob.verifySessionAndConfig(sc, config)
+    SnappyJobValid()
+  }
+
+
+  def createPartitionedRowTableUsingAPI(snSession: SnappySession): Unit = {
+    pw.println()
+    pw.println(s"Creating tables $colTable and $rowTable using API")
+
+    // drop the table if it exists
+    snSession.dropTable(colTable, ifExists = true)
+    snSession.dropTable(rowTable, ifExists = true)
+
+    val schema = StructType(Array(StructField("PS_PARTKEY", IntegerType, false),
+      StructField("S_SUPPKEY", IntegerType, false),
+      StructField("PS_AVAILQTY", IntegerType, false),
+      StructField("PS_SUPPLYCOST", DecimalType(15, 2), false)
+    ))
+
+    val props1 = Map("PARTITION_BY" -> "PS_PARTKEY")
+    snSession.createTable(colTable, "column", schema, props1)
+    snSession.createTable(rowTable, "row", schema, props1)
+
+    val data = Seq(Seq(100, 1, 5000, BigDecimal(100)),
+      Seq(200, 2, 50, BigDecimal(10)),
+      Seq(300, 3, 1000, BigDecimal(20)),
+      Seq(400, 4, 200, BigDecimal(30))
+    )
+    val rdd = snSession.sparkContext.parallelize(data,
+      data.length).map(s => new Data(s(0).asInstanceOf[Int],
+      s(1).asInstanceOf[Int],
+      s(2).asInstanceOf[Int],
+      s(3).asInstanceOf[BigDecimal]))
+
+    val dataDF = snSession.createDataFrame(rdd)
+    pw.println(s"Inserting data in $colTable table")
+    dataDF.write.insertInto(colTable)
+    pw.println(s"Inserting data in $rowTable table")
+    dataDF.write.insertInto(rowTable)
+
+    pw.println(s"Printing the contents of the $colTable table")
+    var tableData = snSession.sql(s"SELECT * FROM $colTable").collect()
+    tableData.foreach(pw.println)
+    pw.println(s"Printing the contents of the $rowTable table")
+    tableData = snSession.sql(s"SELECT * FROM $rowTable").collect()
+    tableData.foreach(pw.println)
+
+    pw.println()
+    pw.println("Update the available quantity for PARTKEY 100")
+    snSession.update(rowTable, "PS_PARTKEY = 100", Row(50000), "PS_AVAILQTY")
+
+    pw.println(s"Printing the contents of the $rowTable table after update")
+    tableData = snSession.sql(s"SELECT * FROM $rowTable").collect()
+    tableData.foreach(pw.println)
+
+    pw.println()
+    pw.println("Delete the records for PARTKEY 400")
+    snSession.delete(rowTable, "PS_PARTKEY = 400")
+
+    pw.println(s"Printing the contents of the $rowTable table after delete")
+    tableData = snSession.sql(s"SELECT * FROM $rowTable").collect()
+    tableData.foreach(pw.println)
+
+    pw.println("****Done****")
+  }
```

Comment:
test code in product class?

Reply:
Did not understand this. This job class is under src/dunit/ so it's not a product class, though this is modified copy of CreateReplicatedRowTable job in examples folder.
May be I should add explicit verification instead of logging to a file?

Reply:
Oh missed that this is a test class. Looked like product code.